src/odinapi/utils/datamodel.py

```diff
@@ -0,0 +1,432 @@
+    d1 = ["time"]
+    d2 = ["time", "level"]
+    d3 = ["time", "level", "level"]
+
+
+class Units(Enum):
+    time = "days since 1858-11-17 00:00"
+    altitude = "m"
+    lat = "degrees north"
+    lon = "degrees east"
+    hours = "hours"
+    unitless = "-"
+    pressure = "Pa"
+    temperature = "K"
+    degrees = "degrees"
+    koverk = "K/K"
+    poverp = "%/%"
+    product = "product"
+
+
+@attr.s
+class Filter:
+    residual = attr.ib(type=float)
+    minlmfactor = attr.ib(type=float)
+
+
+@attr.s
+class Parameter:
+    name = attr.ib(type=str)
+    units = attr.ib(type=Units)
+    description = attr.ib(type=str)
+    dtype = attr.ib(type=DType)
+    dimension = attr.ib(type=Dimension)
+    l2type = attr.ib(type=L2Type)
+
+    def get_description(self, product: str) -> str:
+        if self.name == "Profile":
+            return (
+                "Retrieved temperature profile."
+                if is_temperature(product)
+                else "Retrieved volume mixing ratio."
+            )
+        return self.description
+
+    def get_units(self, product: str) -> Units:
+        if self.name == "AVK":
+            return Units.koverk if is_temperature(product) else Units.poverp
+        elif self.units != Units.product:
+            return self.units
+        else:
+            return (
+                Units.temperature if is_temperature(product)
+                else Units.unitless
+            )
+
+
+@attr.s
+class L2File:
+    parameters = attr.ib(type=List[Parameter])
+
+
+@attr.s
+class L2anc:
+    LST = attr.ib(type=float)
+    Orbit = attr.ib(type=int)
+    SZA1D = attr.ib(type=float)
+    SZA = attr.ib(type=List[float])
+    Theta = attr.ib(type=List[float])
+
+
+@attr.s
+class L2:
+    InvMode = attr.ib(type=str)
+    ScanID = attr.ib(type=int)
+    Time = attr.ib(type=dt.datetime)
+    Lat1D = attr.ib(type=float)
+    Lon1D = attr.ib(type=float)
+    Quality = attr.ib(type=float)
+    Altitude = attr.ib(type=List[float])
+    Pressure = attr.ib(type=List[float])
+    Profile = attr.ib(type=List[float])
+    Latitude = attr.ib(type=List[float])
+    Longitude = attr.ib(type=List[float])
+    Temperature = attr.ib(type=List[float])
+    ErrorTotal = attr.ib(type=List[float])
+    ErrorNoise = attr.ib(type=List[float])
+    MeasResponse = attr.ib(type=List[float])
+    Apriori = attr.ib(type=List[float])
+    VMR = attr.ib(type=List[float])
+    AVK = attr.ib(type=List[List[float]])
+
+
+@attr.s
+class L2i:
+    GenerationTime = attr.ib(type=dt.datetime)
+    Residual = attr.ib(type=float)
+    MinLmFactor = attr.ib(type=float)
+    FreqMode = attr.ib(type=int)
+
+    @property
+    def filter(self) -> Filter:
+        return Filter(
+            residual=1.5,
+            minlmfactor=10. if self.FreqMode in [8., 13., 19.] else 2.
+        )
+
+    def isvalid(self) -> bool:
+        return (
+            self.Residual <= self.filter.residual
+            and self.MinLmFactor <= self.filter.minlmfactor
```

Comment:
We know these values may be other than finite, and operations like less may have unexpected outcomes then.
if a `-Inf` really valid?

Reply:
will add check that they are finite